src/sas/qtgui/Perspectives/Fitting/ConstraintWidget.py

```diff
@@ -492,6 +492,43 @@ def fitComplete(self, result):
             msg = "Fitting failed. Please ensure correctness of chosen constraints."
             self.parent.communicate.statusBarUpdateSignal.emit(msg)
             return
+        # Warn the user if fitting has been unsuccessfull
+        if not result[0][0][0].success:
```

Comment:
A few lines after this if statement the line `results = result[0][0]` appears. Using this definition here could help with the code readability.